File: Upcoder/Other/check số hoàn thiện/LTCB-30.cs
using System;
namespace CheckSHT
{
    public class Program
    {
        static void Main(string[]args)
        {
            int n = int.Parse(Console.ReadLine());
            if(CheckSHT(n))
                Console.Write("YES");
            else
                Console.Write("NO");
        }
        static bool CheckSHT(int x)
        {
            if (x <= 1) 
                return false; 
            int tong = 0;
            for (int i = 1; i <= x / 2; i++) {
                if (x % i == 0) {
                    tong += i;
                }
            }
            return tong == x;
        }
    }
}

File: Upcoder/Other/check số hoàn thiện/LTCB-30.py
def check_sht(x):
    if x <= 1:
        return False
    tong = 0
    for i in range(1, x // 2 + 1):
        if x % i == 0:
            tong += i
    return tong == x

if __name__ == "__main__":
    n = int(input())
    if check_sht(n):
        print("YES")
    else:
        print("NO")
